src/python/pants/backend/python/util_rules/ancestor_files.py

```diff
@@ -36,19 +27,19 @@ class AncestorFiles:
     snapshot: Snapshot
 
 
-def identify_missing_ancestor_files(name: str, sources: Sequence[str]) -> FrozenOrderedSet[str]:
+def putative_ancestor_files(input_files: tuple[str, ...], requested: tuple[str, ...]) -> set[str]:
     """Return the paths of potentially missing ancestor files.
 
     NB: The sources are expected to not have had their source roots stripped.
     Therefore this function will consider superfluous files at and above the source roots,
     (e.g., src/python/<name>, src/<name>). It is the caller's responsibility to filter these
     out if necessary.
     """
-    packages: Set[str] = set()
-    for source in sources:
-        if not source.endswith(".py"):
+    packages: set[str] = set()
+    for input_file in input_files:
+        if not input_file.endswith((".py", ".pyi")):
```

Comment:
This was the main bug

Reply:
Should this list be part of the `AncestorFilesRequest` object?

Reply:
Probably not because this is defined in backend/python